RM ANOVA Simple Effects

JohnnyDoorn opened the pull request:
Fix  #1957

vankesteren commented on JASP-Engine/JASP/R/anovarepeatedmeasures.R:
```diff
@@ -2501,6 +2527,273 @@ AnovaRepeatedMeasures <- function(dataset=NULL, options, perform="run", callback
+      # (because each subset only has one level of that factor). The same procedure is applied to the second moderator, if specified.
+      
+      subOptions <- options
+      if (isModeratorOneWithin) {
```
Yeah alright this if/else block is very long. Do you think it would be shorter with a [`switch()`](https://stackoverflow.com/questions/10393508/how-to-use-the-switch-statement-in-r-functions) statement? If not, leave it like this but maybe add some comments within each condition to show which case we're in.

AlexanderLyNL commented on JASP-Engine/JASP/R/anovarepeatedmeasures.R:
```diff
@@ -2501,6 +2527,273 @@ AnovaRepeatedMeasures <- function(dataset=NULL, options, perform="run", callback
 
+.rmAnovaSimpleEffects <- function(dataset, options, perform, fullModel, fullAnovaTableWithin, fullAnovaTableBetween, status, singular, stateSimpleEffects) {
+  
+  if (identical(options$simpleFactor, "") | identical(options$moderatorFactorOne, ""))
```
Please call a items within a list using "[[]]" rather than with $, for instance, 

options[["simpleFactor"]] 

This is more robust, which is necessary due to the fact that you have items that are named similar such as "moderatorFactorOne" and "moderatorFactorTwo".

AlexanderLyNL commented on JASP-Engine/JASP/R/anovarepeatedmeasures.R:
```diff
@@ -2501,6 +2527,273 @@ AnovaRepeatedMeasures <- function(dataset=NULL, options, perform="run", callback
+                                      FUN = function(x){x$name == options$moderatorFactorOne}) == TRUE)
+        splitNames <- unlist(lapply(strsplit(factorNamesV,  split = "_"), 
+                                      FUN = function(x) x[indexofOrderFactors[rmFactorIndex]]))
+        splitWithinNames <- unlist(lapply(strsplit(withinFactorNamesV,  split = "_"), 
```
Perhaps we should make standard methods for this operation in common

JohnnyDoorn commented:
I have addressed the comments in the latest commits. Also added functionality to either pool the error terms or not - the literature seems to disagree on this, so I figured I put it in as an option. The default is set to non-pooled, as this is more robust to violations of homogeneity of variances.
